app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/event_screen/components/PhotoModal.kt

```diff
@@ -0,0 +1,181 @@
+import androidx.compose.material.icons.outlined.Close
+import androidx.compose.material.icons.outlined.Delete
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
+import com.realityexpander.tasky.R
+import com.realityexpander.tasky.agenda_feature.domain.Photo
+import com.realityexpander.tasky.core.presentation.common.modifiers.DP
+import com.realityexpander.tasky.core.presentation.theme.TaskyTheme
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun PhotoModal(
+    title: String,
+    photo: Photo,
+    isRemoveEnabled: Boolean = false,
+    onRemove: () -> Unit,
+    onCancel: () -> Unit,
+) {
+
+    var isConfirmRemoveDialogVisible by remember { mutableStateOf(false) }
+
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(Color.Black.copy(alpha = .5f))
+    ) {
+
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .background(MaterialTheme.colors.surface)
+
+        ) {
+
+            Row(
+                horizontalArrangement = Arrangement.SpaceBetween,
+                verticalAlignment = Alignment.CenterVertically,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(DP.small)
+            ) {
+
+                // Close button
+                Icon(
+                    Icons.Outlined.Close,
+                    tint = MaterialTheme.colors.onSurface,
+                    contentDescription = stringResource(R.string.back_description),
+                    modifier = Modifier
+                        .size(32.dp)
+                        .align(Alignment.CenterVertically)
+                        .clickable { onCancel() }
+                )
+
+                // Title
+                Text(
+                    title,
+                    color = MaterialTheme.colors.onSurface,
+                    style = MaterialTheme.typography.h4,
+                    textAlign = TextAlign.Start,
+                    modifier = Modifier
+                )
+
+                // Remove button
+                Icon(
+                    Icons.Outlined.Delete,
+                    tint = if (isRemoveEnabled) MaterialTheme.colors.onSurface else Color.Transparent,
+                    contentDescription = stringResource(R.string.remove_description),
+                    modifier = Modifier
+                        .size(32.dp)
+                        .align(Alignment.CenterVertically)
+                        .clickable(isRemoveEnabled) {
+                            isConfirmRemoveDialogVisible = true
+                        }
+                )
```

Comment:
If you use clickable icons, I'd prefer to use `IconButton` instead which comes with all the default ripples and padding all icon buttons have on Android according to Material Design. It's something users are used to

Reply:
ok i didnt know about that one. Will swap out.